sdk/python/kfp/dsl/for_loop.py

```diff
@@ -191,7 +194,7 @@ def name_is_loop_argument(cls, name: str) -> bool:
           or (cls.LOOP_ITEM_PARAM_NAME_BASE + '-') in name
 
 
-class LoopArgumentVariable(pipeline_channel.PipelineChannel):
+class LoopArgumentVariable(pipeline_channel.PipelineParameterChannel):
```

Comment:
This line is the fix. This is the right thing, since we don't support iteration over artifacts, so a `LoopArgumentVariable` will always be a `PipelineParameterChannel` and cannot be a `PipelineArtifactChannel`.